src/EnergyPlus/WindowManagerExteriorOptical.cc

```diff
@@ -0,0 +1,645 @@
+    // 		auto& construction( Construct( ConstrNum ) );
+    // 		if ( construction.isGlazingConstruction() ) {
+    // 			for ( auto LayNum = 1; LayNum <= construction.TotLayers; ++LayNum ) {
+    // 				auto& material( Material( construction.LayerPoint( LayNum ) ) );
+    // 				if ( material.Group != WindowGas && material.Group != WindowGasMixture &&
+    // 					material.Group != ComplexWindowGap && material.Group != ComplexWindowShade ) {
+    // 					auto aMaterial = std::make_shared< MaterialProperties >();
+    // 					*aMaterial = material;
+    //
+    // 					// This is necessary because rest of EnergyPlus code relies on TransDiff property
+    // 					// of construction. It will basically trigger Window optical calculations if this
+    // 					// property is >0.
+    // 					construction.TransDiff = 0.1;
+    //
+    // 					auto aRange = WavelengthRange::Solar;
+    // 					auto aSolarLayer = getBSDFLayer( aMaterial, aRange );
+    // 					aWinConstBSDF.pushBSDFLayer( aRange, ConstrNum, aSolarLayer );
+    //
+    // 					aRange = WavelengthRange::Visible;
+    // 					auto aVisibleLayer = getBSDFLayer( aMaterial, aRange );
+    // 					aWinConstBSDF.pushBSDFLayer( aRange, ConstrNum, aVisibleLayer );
+    // 				}
+    //
+    // 			}
+    // 		}
+    // 	}
+    // }
+
+    void InitWCE_SimplifiedOpticalData()
+    {
+        // SUBROUTINE INFORMATION:
+        //       AUTHOR         Simon Vidanovic
+        //       DATE WRITTEN   May 2017
+        //       MODIFIED       na
+        //       RE-ENGINEERED  na
+
+        // PURPOSE OF THIS SUBROUTINE:
+        // Initialize scattering construction layers in Solar and Visible spectrum.
+
+        // Calculate optical properties of blind-type layers entered with MATERIAL:WindowBlind
+        // Calculation from this is used for IR properties. Need to make sure that properties
+        // are calculated with new WCE optical engine (for both blinds and screens)
+        if (TotBlinds > 0) CalcWindowBlindProperties();
+
+        // Initialize SurfaceScreen structure
+        NumSurfaceScreens = TotScreens;
+        if (NumSurfaceScreens > 0) CalcWindowScreenProperties();
+
+        auto aWinConstSimp = CWindowConstructionsSimplified::instance();
+        for (auto ConstrNum = 1; ConstrNum <= TotConstructs; ++ConstrNum) {
+            auto &construction(Construct(ConstrNum));
+            if (construction.isGlazingConstruction()) {
+                for (auto LayNum = 1; LayNum <= construction.TotLayers; ++LayNum) {
+                    auto &material(Material(construction.LayerPoint(LayNum)));
+                    if (material.Group != WindowGas && material.Group != WindowGasMixture && material.Group != ComplexWindowGap &&
+                        material.Group != ComplexWindowShade) {
+                        auto aMaterial = std::make_shared<MaterialProperties>();
+                        *aMaterial = material;
```

Comment:
@mbadams5 it calls operator=() to assign data from one object to another

Reply:
I understand that it calls the assignment operator, but what I meant is who owns that data at that point. For example, I know unique_ptr expects to completely own the underlying data (pointer) and will clean up that resource when it falls out of scope. What I am unsure about is whether shared_ptr behaves similarly when the reference count goes to 0, I believe it does though.

Additionally if the Material() isn't created with `new` and the shared_ptr falls out of scope with 0 references, it will probably throw an exception since it doesn't know how to `delete` something not created by `new`.

Reply:
@mbadams5 @jasondegraw You are talking of this issue? I think it is ok, but to be on safe side I think shared_ptr can be removed here. Will take a look later today if I can remove it. Should not be the problem.